src/providers/vehicle-check/vehicle-check.ts

```diff
@@ -0,0 +1,112 @@
+import { Subject } from 'rxjs/Subject';
+import { FaultStoreProvider } from './../fault-store/fault-store';
+import { Injectable } from '@angular/core';
+import 'rxjs/add/observable/of';
+export interface IVehicleCheck {
+  complete?: boolean;
+  faultType?: string;
+}
+
+export enum vCheckType {
+  SHOWME = 'showMe',
+  TELLME = 'tellMe'
+}
+
+@Injectable()
+export class VehicleCheckProvider {
+  private tellMe: IVehicleCheck = {};
+  private showMe: IVehicleCheck = {};
+  showMeSub: Subject<IVehicleCheck>;
+
+  constructor(private faultStore: FaultStoreProvider) {
+    this.showMeSub = new Subject();
+    this.showMeSub.next({ ...this.showMe });
+  }
+
+  /**
+   * Logic to determine which sub-fault should be used to write to the main vehiclecheck fault in state
+   */
+  overwriteFaultWith() {
+    const { faultType: tellMeFault } = this.tellMe;
+    const { faultType: showMeFault } = this.showMe;
+    if (
+      tellMeFault === undefined ||
+      tellMeFault === showMeFault ||
+      (tellMeFault === 'fault' && (showMeFault === 'serious' || showMeFault === 'dangerous')) ||
+      (tellMeFault === 'serious' && showMeFault === 'dangerous')
+    ) {
+      return this.showMe.faultType;
+    }
+
+    return this.tellMe.faultType;
```

Comment:
Can we simplify this with early returns, something like:

if (tellMeFault === 'dangerous' || showMeFault === 'dangerous')
  return 'dangerous';

if (tellMeFault === 'serious' || showMeFault === 'serious')
  return 'serious';

if (tellMeFault === 'fault' || showMeFault === 'fault')
  return 'fault';

return null; (?)

Reply:
updated to be more specific and quickest first